pkg/csi/service/common/vsphereutil.go

```diff
@@ -1057,6 +1057,34 @@ func DeleteSnapshotUtil(ctx context.Context, manager *Manager, csiSnapshotID str
 	return nil
 }
 
+// GetCnsVolumeType is the helper function that determines the volume type based on the volume-id
+func GetCnsVolumeType(ctx context.Context, manager *Manager, volumeId string) (string, error) {
+	log := logger.GetLogger(ctx)
+	var volumeType string
+	queryFilter := cnstypes.CnsQueryFilter{
+		VolumeIds: []cnstypes.CnsVolumeId{{Id: volumeId}},
+	}
+	querySelection := cnstypes.CnsQuerySelection{
+		Names: []string{
+			string(cnstypes.QuerySelectionNameTypeVolumeType),
+		},
+	}
+	// Select only the volume type.
+	queryResult, err := manager.VolumeManager.QueryAllVolume(ctx, queryFilter, querySelection)
+	if err != nil {
+		return "", logger.LogNewErrorCodef(log, codes.Internal,
+			"queryVolume failed for volumeID: %q with err=%+v", volumeId, err)
+	}
+
+	if len(queryResult.Volumes) == 0 {
+		return "", logger.LogNewErrorCodef(log, codes.Internal,
+			"queryVolume failed for volumeID: %q with err=%+v", volumeId, err)
+	}
```

Comment:
So this is like volume not found?  When deleting volume, volume not found should be treated as a success.

Reply:
Done.